sample/src/androidTest/java/com/schibsted/spain/barista/sample/ListsChildClickTest.java

```diff
@@ -23,94 +26,109 @@
-  public void clickRecyclerItemChild() throws Exception {
+  public void clickRecyclerItemChild() {
     openActivity(ListsActivity.buildIntent()
         .withRecyclers(R.id.recycler)
     );
 
     clickListItemChild(20, R.id.yes);
 
     assertResult("yes");
+    spyFailureHandlerRule.assertNoEspressoFailures();
   }
 
   @Test
-  public void clickListViewItemChild() throws Exception {
+  public void clickListViewItemChild() {
     openActivity(ListsActivity.buildIntent()
         .withComplexLists(R.id.listview)
     );
 
     clickListItemChild(20, R.id.yes);
 
     assertResult("yes");
+    spyFailureHandlerRule.assertNoEspressoFailures();
   }
 
-
   @Test
-  public void clickGridViewItemChild() throws Exception {
+  public void clickGridViewItemChild() {
     openActivity(ListsActivity.buildIntent()
-            .withComplexGrids(R.id.gridview)
+        .withComplexGrids(R.id.gridview)
     );
 
     clickListItemChild(20, R.id.yes);
 
     assertResult("yes");
+    spyFailureHandlerRule.assertNoEspressoFailures();
   }
+
   @Test
-  public void clickMultipleRecyclerItemChild_byId() throws Exception {
+  public void clickMultipleRecyclerItemChild_byId() {
     openActivity(ListsActivity.buildIntent()
         .withRecyclers(R.id.recycler, R.id.recycler2)
     );
 
     clickListItemChild(R.id.recycler, 20, R.id.yes);
 
     assertResult("yes");
+    spyFailureHandlerRule.assertNoEspressoFailures();
   }
 
   @Test
-  public void clickMultipleListViewItemChild_byId() throws Exception {
+  public void clickMultipleListViewItemChild_byId() {
     openActivity(ListsActivity.buildIntent()
         .withComplexLists(R.id.listview, R.id.listview2)
     );
 
     clickListItemChild(R.id.listview, 20, R.id.yes);
 
     assertResult("yes");
+    spyFailureHandlerRule.assertNoEspressoFailures();
   }
 
   @Test
-  public void clickMultipleGridViewItemChild_byId() throws Exception {
+  public void clickMultipleGridViewItemChild_byId() {
     openActivity(ListsActivity.buildIntent()
-            .withComplexGrids(R.id.gridview, R.id.gridview2)
+        .withComplexGrids(R.id.gridview, R.id.gridview2)
     );
 
     clickListItemChild(R.id.gridview, 20, R.id.yes);
 
     assertResult("yes");
+    spyFailureHandlerRule.assertNoEspressoFailures();
   }
 
-  @Test(expected = PerformException.class)
-  public void fails_whenListViewChildNotExist() throws Exception {
+  @Test
+  public void fails_whenRecyclerChildNotExist() {
     openActivity(ListsActivity.buildIntent()
-        .withComplexLists(R.id.listview)
+        .withRecyclers(R.id.recycler)
     );
 
-    clickListItemChild(20, R.id.unknown);
+    Throwable thrown = catchThrowable(() -> clickListItemChild(20, R.id.unknown));
 
-    assertResult("");
+    spyFailureHandlerRule.assertEspressoFailures(1);
+    assertThat(thrown).isInstanceOf(BaristaException.class)
+        .hasMessageContaining("Could not perform action (actionOnItemAtPosition performing ViewAction: Click on a child view ")
```

Comment:
Doesn't it is a little too flaky?

Reply:
Yep. Or even if we update Espresso and it changes something in the message format. 
But I think the benefit of having the full message here is greater than the cost. Plus it serves as documentation of the messages from Barista. Previously we had no idea of what it would show, and some of them were actually wrong.

Also, if we see they fail too often we can always change them for more generic checks.

Reply:
Ok you got me! 🏆